### PR TITLE
build(toolkit): enforce @internal in published types via post-build member stripping

### PR DESCRIPTION
**Pull request by the-ult**
Makes `@internal` enforced in the published types instead of advisory — but not via the compiler flag the issue title suggests, because that flag demonstrably cannot work here.

**Why not `stripInternal`:** enabling it breaks ng-packagr's multi-entry-point `.d.ts` rollup outright. Reproduced twice independently (implementation and review): with the flag on, untagged public symbols (`NGX_SIGNAL_FORMS_CONFIG`, `NgxFieldIdentity`, `NGX_SIGNAL_FORM_CONTEXT`, ...) vanish from the bundled declarations even after removing every `@internal` tag in the package; toggling the flag off restores them immediately. The failure is the flag × rollup interaction, not tag placement — no amount of per-symbol untagging fixes it. `tsconfig.lib.json` stays untouched, and the README documents why.

**What enforces it instead:** a new post-build script, `strip-internal-members.mjs`, running before the existing `strip-internal-exports.mjs`. It deletes `@internal`-tagged declarations nested inside class/interface bodies from the emitted `.d.ts` bundles — the actual leak: `NgxFieldIdentity`'s five `set*` writers (core), `hintChildren`/`hintDescriptors` (form-field), and `connectFieldState` (headless); the build log reports exactly which. Top-level `@internal` declarations in `/core` are deliberately left alone: they are build-time plumbing for sibling entries and `libs/debugger`, and are already unreachable externally because `strip-internal-exports.mjs` removes `"./core"` from the published exports map.

**Safety net:** the scan is a line-based bracket counter, so after stripping, every rewritten file is re-parsed with the TypeScript compiler API (`ts.createSourceFile` parse diagnostics) and the build fails loudly — original file untouched — on any syntax error. One test constructs an input that genuinely fools the counter (unbalanced brace inside a string-literal type) and proves the guard trips instead of publishing corrupted output. Eight tests total, including a template-literal-type false-positive guard; error reporting follows the sibling script's `[toolkit] ERROR` + non-zero-exit convention.

Fixes #289

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Published TypeScript declarations now exclude internal members nested within classes, interfaces, and namespaces.
  * Public APIs and top-level declarations remain preserved and valid.
  * Added safeguards to detect malformed declarations and prevent publishing corrupted output.

* **Documentation**
  * Clarified how internal members are removed from published declarations and why existing TypeScript declaration settings remain unchanged.

* **Tests**
  * Added coverage for nested member removal, declaration preservation, edge cases, and publishing failures.

### CODE REVIEW

File: packages/toolkit/scripts/strip-internal-members.mjs [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+import { readFileSync, readdirSync, writeFileSync } from 'node:fs';
+import { join, resolve } from 'node:path';
+import { pathToFileURL } from 'node:url';
+import ts from 'typescript';
+
+// A real `@internal` JSDoc tag stands alone on its comment line (optionally
+// followed by trailing whitespace only). Prose that references `@internal`
+// inside a sentence always has more text after it on the same line.
+const INTERNAL_TAG_LINE = /^\s*\*\s*@internal\s*$/u;
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/toolkit/scripts/strip-internal-members.mjs]: Accepted — fixed in f2610c4d: the scan is rewritten on the TypeScript AST; @internal detection now uses ts.getJSDocTags, which natively recognizes the one-line /** @internal */ form (new dedicated test).

File: packages/toolkit/scripts/strip-internal-members.mjs [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+ * @param {string} line
+ * @returns {number}
+ */
+function bracketDelta(line) {
+  let delta = 0;
+  for (const ch of line) {
+    if (ch === '{' || ch === '(') delta++;
+    else if (ch === '}' || ch === ')') delta--;
+  }
+  return delta;
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/toolkit/scripts/strip-internal-members.mjs]: Accepted — fixed in f2610c4d: no more character counting — membership is decided by AST parent kind and removals are source spans, so brace-containing string-literal types cannot mis-bound a declaration (test asserts correct output for exactly this shape).

File: packages/toolkit/scripts/strip-internal-members.mjs [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  const touchedFiles = [];
+
+  for (const name of dtsFiles) {
+    const filePath = join(typesDir, name);
+    const content = readFileSync(filePath, 'utf8');
+    const { text, removedCount } = stripNestedInternalMembers(content);
+    if (removedCount === 0) continue;
+
+    assertNoNestedInternalTagsRemain(text, name);
+    assertValidSyntax(text, name);
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/toolkit/scripts/strip-internal-members.mjs]: Accepted — fixed in f2610c4d: the leftover-tag assertion now re-parses and re-walks the AST unconditionally on every file, before any removedCount early-exit.

File: packages/toolkit/scripts/strip-internal-members.mjs [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+      `${start + 1} — bracket depth never returned to ${baseDepth}. ` +
+      `Refusing to guess; fix the parser or the declaration shape.`,
+  );
+}
+
+/**
+ * Removes `@internal`-tagged declarations that are nested inside a class or
+ * interface body (i.e. appear at bracket depth > 0) from `content`. Returns
+ * the rewritten text and the count of declarations removed.
+ *
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/toolkit/scripts/strip-internal-members.mjs:156]: Accepted — fixed in f2610c4d: the doc comment and README now state exactly what is stripped — members nested in class/interface/namespace bodies — matching the AST-based behavior.
